fix(overlayfs): allow hostonly and nfs support

LaszloGombos opened the pull request:
Check for overlay kernel module support.

Execute mount-overlayfs later in the boot process. Without this change `dracut  -a "overlayfs nfs"` does not work as expected.

## Checklist
- [X] I have tested it locally
- [X] I have reviewed and updated any documentation if relevant
- [X] I am providing new code and test(s) for it

Fixes #

LaszloGombos committed:
fix(overlayfs): allow hostonly

Check for overlay kernel module support.

aafeijoo-suse reviewed (commented):
You need to fix the `shfmt` format:

https://github.com/dracutdevs/dracut/actions/runs/4350868266/jobs/7601984774?pr=2269

The "allow hostonly" part is ok, but it has nothing to do with "Execute mount-overlayfs later in the boot process as well.". You should split that into another commit and give the proper reason for that (why does not it work as expected?).

LaszloGombos committed:
fix(overlayfs): to allow overlay on top of network device (nfs)

Execute mount-overlayfs later in the boot process as a pre-pivot
hook. The existing mount hook is skipped for network boot.

Without this change
`dracut  -a "overlayfs nfs"`
would not work as expected.

Improved the existing test case to test for this scenario.

aafeijoo-suse reviewed (approved):
> I improved the test case to test for this scenario, so now everybody can reproduce and understand the failure and also this test case would guard for future regression of this use-case.

Ok, thanks!
